src/searchModal.js

```diff
@@ -419,13 +433,50 @@ export default function searchModalFactory(config) {
      */
     function buildDataModel(data) {
         if (data.settings) {
+            //save availableColumns to memory
+            availableColumns = data.settings.availableColumns;
             // @todo: use the selected columns instead. It can use a promise as it takes place insise a promise chain
-            data.model = columnsToModel(data.settings.availableColumns);
+            data.model = columnsToModel(availableColumns);
+            dataCache = _.cloneDeep(data);
+            return data;
         } else {
             data.model = columnsToModel(_.values(data.model));
+            dataCache = _.cloneDeep(data);
+            return data;
         }
+    }
+
+    /**
+     * Filters datatble model based on stored selected columns
+     * @param {Object} data data containing available columns and model for datatable
+     * @returns {Promise} promise which resolves with filtered data.model
+     */
+    function filterSelectedColumns(data) {
+        return selectedColumnsStore
+            .getItem(getClassFilterUri())
+            .then(storedSelectedColumnIds => {
+                //let columnsToDisplay;
+                if (storedSelectedColumnIds && storedSelectedColumnIds.length) {
+                    selectedColumns = [...storedSelectedColumnIds];
+                    //columnsToDisplay = data.settings.availableColumns.filter(column =>
+                    // selectedColumns.includes(column.id)
+                    //);
+                } else {
+                    selectedColumns = data.settings.availableColumns.reduce((acc, column) => {
+                        if (column.default) {
+                            acc.push(column.id);
+                        }
+                    }, []);
+                }
 
-        return data;
+                data.model = data.model.filter(column => selectedColumns.includes(column.id));
+
+                return data;
+            })
+            .catch(e => {
+                instance.trigger('error', e);
+                reject(new Error('Error getting selected columns'));
```

Comment:
**suggestion:** Actually, the idea of making the error more accurate is good. But then, it should go with error wrapping. The returned promise is caught later, with another trigger, so no need to double it.
```suggestion
                return Promise.reject(new Error('Error getting selected columns', { cause: e }));
```

Reply:
yes, exactly. But I just did remove that reject code